filters/StacFilter.cpp

```diff
@@ -0,0 +1,215 @@
+    n.add(name, box.maxy);
+    n.add(name, box.maxz);
+}
+
+void StacFilter::extractMetadata(PointTableRef table)
+{
+    std::string stem = FileUtils::stem(m_inputFile);
+    std::string fileExt = FileUtils::extension(m_inputFile);
+
+    //Base STAC object
+    MetadataNode id = m_metadata.add("id", stem);
+    MetadataNode properties = m_metadata.add("properties");
+    properties.add("datetime", "2022-11-15T16:06:02.616469Z");
+    m_metadata.add("type", "Feature");
+    m_metadata.add("stac_version", "1.0.0");
+
+    //extensions - pointcloud and projection extensions
+    m_metadata.add("extensions", "https://stac-extensions.github.io/pointcloud/v1.0.0/schema.json");
+    m_metadata.add("extensions", "https://stac-extensions.github.io/projection/v1.1.0/schema.json");
+
+    //links
+    MetadataNode self = m_metadata.addList("links");
+    self.add("rel", "derived_from");
+    self.add("href", m_inputFile);
+
+    //assets - add source file to data asset
+    MetadataNode assets = m_metadata.add("assets");
+    MetadataNode data;
+    data.add("href", m_inputFile);
+    data.add("title", "Lidar data");
+    assets.add(data.clone("data"));
+
+    //Add dimension statistics
+    uint32_t position(0);
+    point_count_t count = 0;
+    bool bNoPoints(true);
+    for (auto di = m_stats.begin(); di != m_stats.end(); ++di)
+    {
+        Summary& s = di->second;
+
+        bNoPoints = (bool)s.count();
+        count = s.count();
+
+        MetadataNode pcStats = properties.addList("pc:statistics");
+        pcStats.add("position", position++);
+        s.extractMetadata(pcStats);
+    }
+
+    //pointcloud extension
+    properties.add("pc:count", count);
+    properties.add("pc:type", "lidar");
+    properties.add("pc:encoding", fileExt);
+    auto dims = table.layout()->toMetadata().children();
+    for (auto& c: dims)
+        properties.add(c.clone("pc:schemas"));
+
+    //Projection and base geometry/bbox info. Base Bbox and geometry in stac
+    //should be in 4326, and anthing prefaced with 'proj' should be in the
+    //native srs. If there is no srs derived from the file, default it to 4326.
+    auto xs = m_stats.find(Dimension::Id::X);
+    auto ys = m_stats.find(Dimension::Id::Y);
+    auto zs = m_stats.find(Dimension::Id::Z);
+    if (xs != m_stats.end() &&
+        ys != m_stats.end() &&
+        zs != m_stats.end() &&
+        bNoPoints)
+    {
+        BOX3D box(xs->second.minimum(), ys->second.minimum(),
+            zs->second.minimum(), xs->second.maximum(), ys->second.maximum(),
+            zs->second.maximum());
+        Polygon p(box);
+        addBox(properties, "proj:bbox", box);
+
+        MetadataNode projgeom = properties.addWithType("proj:geometry",
+            p.json(), "json", "GeoJSON boundary");
+        SpatialReference ref = table.anySpatialReference();
+
+        // if we don't get an SRS from the PointTableRef,
+        // we won't add another metadata node
+        if (!ref.empty())
```

Comment:
Yeah that seems like a good solution. I wasn't sure how to handle it since `filters.stats` just kinda keeps moving if it's not set. If they don't set a `default_srs` and there isn't one in the data, should it just throw?

Reply:
> If they don't set a `default_srs` and there isn't one in the data, should it just throw?

Yes.